collector/lib/CollectorConfig.h

```diff
@@ -105,7 +105,7 @@ end
 
   std::shared_ptr<grpc::Channel> grpc_channel;
 
- private:
+ protected:
```

Comment:
I used this approach because the object seems to be built to be immutable; and I didn't wanted to change that.
I also expect that more members might be subject to change for testing purpose, I making the private section protected was kindof future-proof on this aspect.

Ideally, the configuration object would have getters and setters for all attributes and it would be provided as a const reference in the production code. But I felt that was too big a change. Having remorse now... :)

Reply:
Immutability is a good argument, although I wonder if it was implemented like that on purpose.